Sources/Views/ALKChatCell.swift

```diff
@@ -266,7 +286,29 @@ public final class ALKChatCell: SwipeTableViewCell, Localizable {
     public func update(viewModel: ALKChatViewModelProtocol, identity _: ALKIdentityProtocol?, placeholder: UIImage? = nil) {
         self.viewModel = viewModel
         let placeHolder = placeholderImage(placeholder, viewModel: viewModel)
-
+        
+        /// this is used to set the icon according to data recived in metadata
+        let isAgentApp = ALApplozicSettings.isAgentAppConfigurationEnabled()
+        let channelKey = viewModel.channelKey
+        if isAgentApp == true, let channel = ALChannelService().getChannelByKey(channelKey), let source = channel.metadata.value(forKey: "source") {
+                platformImageView.isHidden = false
+                guard let device = source as? String else {
+                    return
+                }
+                switch device {
+                    case "MOBILE":
+                        platformImageView.image = PlatformIcon.iconPhone
+                    case "WEB":
+                        platformImageView.image = PlatformIcon.iconWeb
+                    case "FACEBOOK":
+                        platformImageView.image = PlatformIcon.iconFacebook
+                    case "WHATSAPPCLOUDAPI":
+                        platformImageView.image = PlatformIcon.iconWhatsapp
+                    default:
+                        platformImageView.isHidden = true
+                    }
```

Comment:
this is not formatted (check the alignment) and check we can create a method to get the respective image for device and set the image in single place instead of calling set image inside the switch statement

Reply:
- made the changes created the dictionary and removed switch. 
- reformatted the code alignment. 
- moved the platformIcon constraints in the condition block as earlier it was creating issue while the device name is not matching with the device key in the dictionary
- did couple of changes in the constraint so that it will not effect the email icon.